src/realm/alloc_slab.cpp

```diff
@@ -924,8 +924,12 @@ ref_type SlabAlloc::attach_file(const std::string& file_path, Config& cfg)
         size = initial_size;
     }
     ref_type top_ref;
+    realm::_impl::On_scope_exit reader_end_guard;
     try {
         File::Map<char> map(m_file_mappings->m_file, File::access_ReadOnly, size); // Throws
+        note_reader_start(this);
+        realm::_impl::On_scope_exit eg( [this]() { note_reader_end(this); } );
+        reader_end_guard = std::move(eg);
```

Comment:
`reader_end_guard = realm::_impl::On_scope_exit( [this]() { note_reader_end(this); } );`
would be shorter.